x-pack/solutions/observability/plugins/observability/public/pages/util/synthetics_alert_params_to_kql.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, type KueryNode, nodeBuilder, toKqlExpression } from '@kbn/es-query';
+import type { SyntheticsMonitorStatusRuleParams } from '@kbn/response-ops-rule-params/synthetics_monitor_status';
+import type { TLSRuleParams } from '@kbn/response-ops-rule-params/synthetics_tls';
+
+const MS_TO_DAYS = 1000 * 60 * 60 * 24;
+
+const supportedAlertParams = [
+  'tags',
+  'locations',
+  'monitorIds',
+  'monitorTypes',
+  'projects',
+] as const;
+
+type SupportedAlertParam = (typeof supportedAlertParams)[number];
+
+const fieldMap: Record<SupportedAlertParam, string> = {
+  locations: 'observer.name',
+  monitorIds: 'monitor.id',
+  monitorTypes: 'monitor.type',
+  projects: 'project.id',
+  tags: 'tags',
+};
+function getFieldnameForKey(key: SupportedAlertParam): string {
+  if (!fieldMap[key]) {
+    throw new Error(`Unsupported alert param: ${key}`);
+  }
```

Comment:
nit: if key is of type `SupportedAlertParam` I think we can avoid this check.
```suggestion
```

Reply:
50a893a50a5ae72815c35a0ab048c700f94593bf